src/string/anagram.rs

```diff
@@ -1,27 +1,101 @@
-pub fn check_anagram(s: &str, t: &str) -> bool {
-    sort_string(s) == sort_string(t)
+use std::collections::HashMap;
+
+/// Custom error type representing an invalid character found in the input.
+#[derive(Debug, PartialEq)]
+pub enum AnagramError {
+    NonAlphabeticCharacter,
 }
 
-fn sort_string(s: &str) -> Vec<char> {
-    let mut res: Vec<char> = s.to_ascii_lowercase().chars().collect::<Vec<_>>();
-    res.sort_unstable();
-    res
+/// Checks if two strings are anagrams, ignoring spaces and case sensitivity.
+///
+/// # Arguments
+///
+/// * `s` - First input string.
+/// * `t` - Second input string.
+///
+/// # Returns
+///
+/// * `Ok(true)` if the strings are anagrams.
+/// * `Ok(false)` if the strings are not anagrams.
+/// * `Err(AnagramError)` if either string contains non-alphabetic characters.
+pub fn check_anagram(s: &str, t: &str) -> Result<bool, AnagramError> {
+    let s_cleaned = clean_string(s)?;
+    let t_cleaned = clean_string(t)?;
+
+    Ok(char_frequency(&s_cleaned) == char_frequency(&t_cleaned))
+}
+
+/// Cleans the input string by removing spaces and converting to lowercase.
+/// Returns an error if any non-alphabetic character is found.
+///
+/// # Arguments
+///
+/// * `s` - Input string to clean.
+///
+/// # Returns
+///
+/// * `Ok(String)` containing the cleaned string (no spaces, lowercase).
+/// * `Err(AnagramError)` if the string contains non-alphabetic characters.
+fn clean_string(s: &str) -> Result<String, AnagramError> {
+    s.chars()
+        .filter(|c| !c.is_whitespace())
+        .map(|c| {
+            if c.is_alphabetic() {
+                Ok(c.to_ascii_lowercase())
+            } else {
+                Err(AnagramError::NonAlphabeticCharacter)
+            }
+        })
+        .collect()
+}
+
+/// Computes the frequency of characters in a string.
+///
+/// # Arguments
+///
+/// * `s` - Input string.
+///
+/// # Returns
+///
+/// * A `HashMap` where the keys are characters and values are their frequencies.
+fn char_frequency(s: &str) -> HashMap<char, usize> {
```

Comment:
```suggestion
fn char_count(s: &str) -> HashMap<char, usize> {
```
It is not the _frequency_.

Reply:
Will rename the function